freud/locality.pyx

```diff
@@ -745,15 +898,12 @@ cdef class AABBQuery(NeighborQuery):
         if points.shape[1] != 3:
             raise TypeError('points should be an Nx3 array')
 
-        # Ensure that enough neighbors are found when excluding
-        if exclude_ii:
-            k += 1
-
         # Default guess value
         if r == 0:
-            r = 0.1*min(self._box.L)
+            r = 0.1*min(
+                [length for length in self._box.L if length != 0])
```

Comment:
If this is meant to address 2D behavior, I'd like to check the `is2D` property and make that logic explicit, rather than rely on `Lz=0` as this seems to do.